js/navbar/tabColor.js

```diff
@@ -245,7 +200,61 @@ function setColor (bg, fg) {
   }
 }
 
-// theme changes can affect the tab colors
-window.addEventListener('themechange', function (e) {
-  updateColorPalette()
-})
+const tabColor = {
+  initialize: function () {
+    webviews.bindEvent('page-favicon-updated', function (e, favicons) {
+      var id = webviews.getTabFromContents(this)
+      tabColor.updateFromImage(favicons, id)
+    })
+
+    // theme changes can affect the tab colors
+    window.addEventListener('themechange', function (e) {
+      tabColor.refresh()
+    })
+  },
+  updateFromImage: function (favicons, tabId) {
+    // private tabs always use a special color, we don't need to get the icon
+    if (tabs.get(tabId).private === true) {
+      return
+    }
+
+    requestIdleCallback(function () {
+      colorExtractorImage.onload = function (e) {
+        let backgroundColor = getColorFromImage(colorExtractorImage)
```

Comment:
May I express my fondness towards `const`? :-P

Reply:
Done in e30edd7a7d1bd6d5ad57d8431c0ccba7c9f8a325, everything that's constant should be a `const` now.